core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -131,58 +132,127 @@ public static <T> void prepareAuthorizationHooks(T manager, DispatcherHooks hook
       hooks.addPostHook(
           "createMetalake",
           (args, metalake) -> {
-            // TODO: Add the logic of setting the owner
+            NameIdentifier identifier = (NameIdentifier) ((Object[]) args)[0];
+            GravitinoEnv.getInstance()
+                .ownershipManager()
+                .setOwner(
+                    identifier,
+                    Entity.EntityType.METALAKE,
+                    PrincipalUtils.getCurrentUserName(),
+                    Owner.Type.USER);
```

Comment:
We should not put important logic into the util class.

Reply:
I will refactor the code using writing another dispatcher.